plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotTypeConverter.java

```diff
@@ -53,24 +57,39 @@ public PinotTypeConverter(TypeManager typeManager)
 
     public Type toTrinoType(FieldSpec field)
     {
-        return toTrinoType(field.getDataType(), field.isSingleValueField());
+        return toTrinoType(field.getDataType(), field.isSingleValueField(), getFormatSpec(field));
+    }
+
+    private static Optional<DateTimeFormatSpec> getFormatSpec(FieldSpec field)
+    {
+        if (field instanceof DateTimeFieldSpec) {
+            return Optional.of(((DateTimeFieldSpec) field).getFormatSpec());
+        }
+        return Optional.empty();
+    }
+
+    private static boolean isDateType(DateTimeFormatSpec formatSpec)
+    {
+        return formatSpec.getColumnUnit() == TimeUnit.DAYS &&
+                formatSpec.getTimeFormat() == DateTimeFieldSpec.TimeFormat.EPOCH &&
+                formatSpec.getColumnSize() == 1;
```

Comment:
I think we can merge these methods into one:
```java
    private static boolean isDateType(FieldSpec fieldSpec)
    {
        if (fieldSpec instanceof DateTimeFieldSpec dateTimeFieldSpec) {
            DateTimeFormatSpec formatSpec = dateTimeFieldSpec.getFormatSpec();
            return formatSpec.getColumnUnit() == TimeUnit.DAYS &&
                   formatSpec.getTimeFormat() == DateTimeFieldSpec.TimeFormat.EPOCH &&
                   formatSpec.getColumnSize() == 1;
        }
        return false;
    }
...
    private Type toTrinoType(FieldSpec.DataType dataType, boolean isSingleValue, Optional<FieldSpec> formatSpec)
```

Reply:
The fieldspec is not always available, i.e. from a transform function. Also, pinot server data (`DataTable`) uses a different data type `ColumnDataType` vs `Fieldspec.DataType`, that's why there's so many of those `toTrinoType` functions. lmk if I read your suggestion correctly.